src/main/java/com/tutorhub/model/course/Progress.java

```diff
@@ -2,30 +2,54 @@
 
 import jakarta.persistence.Column;
 import jakarta.persistence.Entity;
-import jakarta.persistence.FetchType;
-import jakarta.persistence.GeneratedValue;
-import jakarta.persistence.GenerationType;
 import jakarta.persistence.Id;
+import jakarta.persistence.IdClass;
 import jakarta.persistence.JoinColumn;
-import jakarta.persistence.OneToMany;
+import jakarta.persistence.ManyToOne;
 import jakarta.persistence.Table;
 import lombok.Getter;
 import lombok.Setter;
 
-import java.util.List;
-
 @Entity
 @Table(name = "progress")
 @Getter
 @Setter
+@IdClass(ProgressId.class)
 public class Progress {
 
   @Id
-  @GeneratedValue(strategy = GenerationType.IDENTITY)
-  private Long id;
+  @Column(name = "module_id", nullable = false)
```

Comment:
You can avoid such names properties because they would be set by default as `module_id` and `submodule_id` and `course_info_id`

Reply:
Fixed